tutor/src/screens/question-library/index.jsx

```diff
@@ -19,12 +19,13 @@ class QuestionsDashboardShell extends React.Component {
   }
 
   componentDidMount() {
-    this.course.referenceBook.fetch();
+    if (!this.course.referenceBook.uuid)
+      this.course.referenceBook.fetch();
   }
```

Comment:
You can use `this.course.referenceBook.ensureLoaded()` here.  `ensureLoaded` is part of the base model so it should be on everything

Reply:
@nathanstitt the `ensureLoaded` is working, but it is Promise. I wonder how is that getting evaluated to a Boolean 🤔

Reply:
@camliu89 sorry I was unclear here.  the `.ensureLoaded()` method will load the model if it's not already been loaded.  So you don't need an if at all, it could just be:
```
componentDidMount() {
     this.course.referenceBook.ensureLoaded()
}
```

As you noted it does return a promise but that's just so you can wait on it if you need to.  In the already loaded case it returns just `Promise.resolve()`